staging/src/k8s.io/apiserver/pkg/authorization/authorizer/interfaces.go

```diff
@@ -61,6 +62,9 @@ type Attributes interface {
 
 	// GetPath returns the path of the request
 	GetPath() string
+
+	// GetAuditEvent returns the audit event of the request
+	GetAuditEvent() *audit.Event
```

Comment:
this interface is supposed to be fully serializable so it can be delegated via a subject access review. an audit.Event pointer breaks that

Reply:
Which means that we have to expose the audit annotations (or any other data structure we choose) for authz/n in a different way here. Also a delgated authorizer should fill in values.

Reply:
authz already has the ability to return a reason along with the authorization decision. can we use that?

Reply:
@tallclair @crassirostris can you comment whether the reason is enough?

Reply:
I would expect the following behavior:
* when authorization failed, the reason would contain aggregated reasons from all authorizers that provided reasons
* when authorization succeeded, the reason would contain the reason from the succeeding authorizer if it provided one

Reply:
the authorizer reason is currently plumbed all the way up to https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/filters/authorization.go#L49 which seems like a much easier place to get a handle to the audit event

Reply:
strawman PR that makes RBAC return a detailed reason for what it allows that includes the subject, binding, and role: https://github.com/kubernetes/kubernetes/pull/58531

Reply:
Done.